library/std/src/sys/windows/stdio.rs

```diff
@@ -170,14 +169,27 @@ fn write(
 }
 
 fn write_valid_utf8_to_console(handle: c::HANDLE, utf8: &str) -> io::Result<usize> {
+    debug_assert!(!utf8.is_empty());
```

Comment:
Note that the distributed std is pre-compiled with `debug-assertions` set to false so `debug_assert` will be removed even if the user is doing a debug build. The std can be compiled with debug assertions enabled using a nightly toolchain with the unstable build-std feature.

Reply:
mmh fair - this is not a big deal, since it'll hit a regular assert further down. In debug mode, it'll just give a nicer error.